### PR TITLE
updated tests for children ordinal and syntax kind and etc

### PR DESCRIPTION
**Pull request by heejaechang**
this changes every single operation tests.
fixes https://github.com/dotnet/roslyn/issues/22752

### CODE REVIEW

File: src/Test/Utilities/Portable/CommonTestBase.cs [hunk 1]
```diff
@@ -608,7 +608,27 @@ internal static void VerifyClone(SemanticModel model)
...
+                        if (child == operation)
+                        {
+                            break;
+                        }
+
+                        count++;
+                    }
+
+                    // need to clone from parent since OperationTreeVerifier puts parent information in the output
+                    clonedOperation = model.CloneOperation(operation.Parent).Children.Skip(count).First();
```
Comment (sharwell) [inline src/Test/Utilities/Portable/CommonTestBase.cs]: :bulb: `ElementAt(count)`

Or even better, `First(child => child == operation)` (which eliminates the `count` variable and the entire `foreach` block).

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 2]
```diff
@@ -65,6 +65,9 @@ private void LogCommonPropertiesAndNewLine(IOperation operation)
         {
             LogString(" (");
 
+            // Child Ordial
+            LogString($"[{GetChildOrdinal(operation)}] ");
```
Comment (sharwell) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: :bulb: This would be better implemented as a change on its own. Also, since this change is intended to highlight the behavior change in #22726, initially implement the change against the master branch *before* #22726 is merged, showing the original order. Then merge the current master branch into the PR branch, and update the tests to show the new order.

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 3]
```diff
@@ -100,15 +103,53 @@ private void LogCommonPropertiesAndNewLine(IOperation operation)
                 LogString(", IsImplicit");
             }
 
-
             LogString(")");
 
             // Syntax
-            LogString($" (Syntax: {GetSnippetFromSyntax(operation.Syntax)})");
+            LogString($" (Syntax: {GetKind(operation.Syntax)}, {GetSnippetFromSyntax(operation.Syntax)})");
```
Comment (sharwell) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: :bulb: This would be better implemented as a change on its own. If the change highlights a behavior difference in #22726, it should be done in the same way as the previous comment. Otherwise, if the change is just additional information it can be implemented as a PR against the current master branch. It may be helpful to wait until the previous comment is implemented and merged to avoid conflicts with this work.

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 4]
```diff
@@ -100,15 +103,53 @@ private void LogCommonPropertiesAndNewLine(IOperation operation)
...
+
+            // Parent
+            if (operation == _root)
+            {
+                LogString($" (Parent: {operation.Parent?.Kind})");
```
Comment (sharwell) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: :bulb: This change can be implemented alongside the previous one (showing the current syntax kind).

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 5]
```diff
@@ -306,22 +345,29 @@ private void VisitArrayCommon<T>(ImmutableArray<T> list, string header, bool log
 
         internal void VisitSymbolArrayElement(ISymbol element)
         {
+            Indent();
```
Comment (sharwell) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: :bulb: The indentation change should be implemented as its own separate PR against the master branch, after the other test changes are completed and merged.

File: src/Test/Utilities/Portable/CommonTestBase.cs (hunk 1 above)
Comment (heejaechang) [reply src/Test/Utilities/Portable/CommonTestBase.cs]: 2 are 2 different tree, can't use First(...). sure, I can use ElementAt

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs (hunk 2 above)
Comment (heejaechang) [reply src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: oh.. well.. sure..

**Comment by heejaechang** [comment]
@dotnet/analyzer-ioperation @AlekseyTs can you take a look?

**Comment by heejaechang** [comment]
retest this please

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure not sure what actually failed [windows_build_correctness_prtest ]

Build step 'Execute Windows batch command' marked build as failure

https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_build_correctness_prtest/10138/

**Comment by heejaechang** [comment]
retest windows_build_correctness_prtest  please

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure is "windows_build_correctness_prtest" broken currently?

**Comment by jasonmalinowski** [comment]
@heejaechang It's flagging your removal of a ProjectReference in BasicCompilerSyntaxTest.vbproj. Not sure why there were two references, but something looks wrong there.

**Comment by heejaechang** [comment]
@jasonmalinowski ah, yes, I removed that. there was duplicate p2p reference so MSBuildWorkspace throws an exception (for ioperation internal tool) since MSBuildWorkspace looks like doesn't have guard for duplicated p2p references (VS looks like have the guard).

is the duplicated p2p reference intentional?

**Comment by jasonmalinowski** [comment]
@heejaechang No idea. I suspect you wanted to delete the other one though.

**Comment by heejaechang** [comment]
@jasonmalinowski ah, yes. sorry I deleted right one, and left wrong one in.

**Comment by heejaechang** [comment]
@AlekseyTs prefers not adding this data in the dump but create more targeted tests on Children property.

I am closing this PR and will create new tests focusing only on Children property.
